client-js/app/elements/labrad-nodes.ts

```diff
@@ -331,152 +392,227 @@ export class LabradNodes extends polymer.Base {
   }
 
 
-  /**
-   * Find the index in the node info list of a node with the given name.
-   *
-   * If no such node is found, returns -1.
-   */
-  private _nodeIndex(name: string): number {
-    var idx = -1;
-    for (var i = 0; i < this.info.length; i++) {
-      if (this.info[i].name === name) {
+  private getNodeIndex(item: (NodeStatus | ServerDisconnectMessage), array: NodeStatus[]): number {
+    let idx = -1;
+    for (let i = 0; i < array.length; ++i) {
+      if (array[i].name === item.name) {
         idx = i;
         break;
```

Comment:
To match `getServerIndex`, could ditch `idx`, `return i;` here and then `return -1;` at the bottom.